// Copyright 2011, Tencent Inc.
// Author: CHEN Feng <phongchen@tencent.com>

option java_package="com.soso.poppy";

package poppy;

// Options for RpcClient ctors
message RpcClientOptions {
    // Whether own netframe.
    // used only by the ctors which support user provided non-NULL netframe
    // parameter, in any other case, this field is ignored.
    optional bool own_netframe = 1;

    // Number of work threads for internal created netframe
    // If not set, number of system cores will be used
    optional int32 work_thread_number = 2;

    // RpcClient mentain a internal channel cache for same address to avoid
    // connect/disconnect frequently.
    // If noy set, RpcClient::kDefaultChannelCacheSize will be used
    optional int32 channel_cache_size = 3;

    // Number of threads in thread pool to execute async complete callnacks
    // If not set, 8 will be used defaultly.
    optional int32 callback_thread_number = 4;
}

